### PR TITLE
Analytics button

### PR DESCRIPTION
**Pull request by markonmars**
Merge for Action button overflow menu.

This allows for Action buttons to be specified in the grid constants file. 

The action button section of the grid will show either

- The given button when the action button list size is one.
- An overflow menu button which provides a menu of all the buttons.

### CODE REVIEW

File: src/components/grid/AnalyticsButton.tsx [hunk 1]
```diff
@@ -2,8 +2,10 @@ import { Search } from '@material-ui/icons'
 
 import ActionButton from './ActionButton'
 
-const AnalyticsButton = () => {
-    return <ActionButton actionIcon={<Search style={{ width: '1.2rem' }} />} />
+const AnalyticsButton = ({ }) => {
```
Comment (monte-monstera) [inline src/components/grid/AnalyticsButton.tsx:5]: small nit pick, don't need the { } in the line above (**{ }**) => { 

don't need to change this for the PR though

File: src/components/grid/dropdown/DropdownItem.module.scss [hunk 2]
```diff
@@ -0,0 +1,20 @@
+@import 'variables';
+
+.container {
+    color : #d6cdcd;
```
Comment (monte-monstera) [inline src/components/grid/dropdown/DropdownItem.module.scss:4]: I've been trying to abstract any major colors to the _variables file so easy to globally change, not necessarily, but might want to add this color in there if it's gonna be used elsewhere? see the $primaryTextColor / $secondaryTextColor

File: src/components/grid/dropdown/DropdownItem.module.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    font-size: 0.75rem;
+    height:1.2rem;
+    padding-top: 0.4rem;
+    padding-left:0.9rem;
+    padding-right:0.9rem;
+    padding-bottom:0.4rem;
```
Comment (monte-monstera) [inline src/components/grid/dropdown/DropdownItem.module.scss:10]: ^ you can 1 line these 4 lines as "padding: 0.4rem 0.9rem", again not required :)

File: src/components/grid/dropdown/OverflowDropdown.module.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+@import 'variables';
+
+.container {
+    padding-top: 0.4rem;
+    padding-bottom: 0.4rem;
+    border-radius: 0.3rem;
+    background-color:#252731;
```
Comment (monte-monstera) [inline src/components/grid/dropdown/OverflowDropdown.module.scss:7]: ^ Same with this color, is it a one off or should it be the same as one defined in variables / added as a new one to variables?

$backgroundColor: #1b1d24 !default;
$secondaryBackgrounColor: #282a33 !default;
$tertiaryBackgrounColor: $grey !default;

File: src/constants/dashboardGridConstants.js [hunk 5]
```diff
@@ -1,6 +1,26 @@
 import DepositActions from '../layouts/grid/DepositActions'
 import BorrowActions from '../layouts/grid/BorrowActions'
 import NumberFormat from 'react-number-format';
+import { Input, SwapHoriz, PieChart } from '@material-ui/icons'
```
Comment (monte-monstera) [inline src/constants/dashboardGridConstants.js:4]: ^ Should we be using the new icon pack that is in this ticket? https://trello.com/c/rb3CxbsM/37-update-icons-with-ones-provided-by-mathijs

File: src/layouts/grid/BorrowActions.tsx [hunk 6]
```diff
@@ -1,16 +1,29 @@
+import React from 'react'
+import ActionButton from '../../components/grid/ActionButton'
 import BorrowButton from '../../components/grid/BorrowButton'
-import OverflowButton from '../../components/grid/OverlowButton'
+import OverflowDropdown from '../../components/grid/dropdown/OverflowDropdown'
```
Comment (monte-monstera) [inline src/layouts/grid/BorrowActions.tsx:4]: you can 1 line this also, checkout components/index.ts

you could do the same for components/grid/index.ts and then your import could look like

import { ActionButton, BorrowButton, OverflowButton, OverflowDropdown } from '../../components/grid'

File: src/layouts/grid/BorrowActions.tsx [hunk 7]
```diff
@@ -1,16 +1,29 @@
...
+
     // TODO handle onClicks, pass symbol to handler so it can do appropriate actions.
     return (
         <div style={{ display: 'flex', alignItems: 'center' }}>
             <BorrowButton />
-            <OverflowButton />
+            {
+                requiresDropdown
+                    ? <OverflowDropdown menuItems={menuItems} />
+                    : <ActionButton
```
Comment (monte-monstera) [inline src/layouts/grid/BorrowActions.tsx:22]: nice one

File: src/layouts/grid/DepositActions.tsx [hunk 8]
```diff
@@ -1,16 +1,24 @@
 import DepositButton from '../../components/grid/DepositButton'
-import OverflowButton from '../../components/grid/OverlowButton'
+import ActionButton from '../../components/grid/ActionButton'
+import OverflowDropdown from '../../components/grid/dropdown/OverflowDropdown'
```
Comment (monte-monstera) [inline src/layouts/grid/DepositActions.tsx:3]: same thing I mentioned elsewhere about it being possible to one line these imports

File: src/constants/dashboardGridConstants.js (hunk 5 above)
Comment (markonmars) [reply src/constants/dashboardGridConstants.js:4]: Ya should be those are just placeholders until I got the correct icons

File: src/layouts/grid/DepositActions.tsx [hunk 9]
```diff
@@ -1,16 +1,24 @@
...
 
 interface Props {
-    symbol: string
+    symbol: string,
+    menuItems: DropdownItemProps[]
 }
```
Comment (monte-monstera) [inline src/layouts/grid/DepositActions.tsx:9]: are these layouts different enough to be separate layouts? DepositActions.tsx and BorrowActions.tsx? (they might be I didn't look very hard)

**Review by monte-monstera — APPROVED** [review]
Nice work

File: src/layouts/grid/DepositActions.tsx (hunk 9 above)
Comment (markonmars) [reply src/layouts/grid/DepositActions.tsx:9]: Yea they could probably be generalised into a grid actions layout the only thing that is different now is the DepositButton / BorrowButton components

File: src/components/grid/dropdown/DropdownItem.module.scss (hunk 2 above)
Comment (markonmars) [reply src/components/grid/dropdown/DropdownItem.module.scss:4]: Need to check against the hex code in the designs but it looked different to everything else just by eyeballing

**Pull request merged**